import "goby/moos/protobuf/goby_moos_app.proto";
import "goby/common/protobuf/option_extensions.proto";

package multihop;

message UnpackagerConfig
{
  optional GobyMOOSAppConfig common = 1 [(goby.field).description="Configuration common to all Goby MOOS applications"]; // see goby_moos_app.proto
  repeated UnpackagerFieldSpecifics array = 2;
  repeated string load_protobuf_shared_lib = 3;
}

message UnpackagerFieldSpecifics
{
  required string class_name = 1;
  required string moos_var = 2; // the string with which the field is associated
}